### PR TITLE
audio session chart

### PR DESCRIPTION
**Pull request by AnkRaiza**
Relates #450 

Changes in this pull request include:
- New audio player section and chart

screen shot 2017-11-03 at 4 40 32 pm

### CODE REVIEW

**Comment by jkelvie** [comment]
For the chart,the time should be translated into minutes and seconds on the y-axis. The chart should be titled - "Average Session Duration".

I would put average duration and number of sessions as fields above the chart (with the field name and the value to the right of it). Average duration should also be translated into minutes and seconds here.

There should be an additional chart for number of sessions per day. This should be title "Sessions Per Day".

**Comment by jkelvie** [comment]
This is looking close - I mean for the overall numbers for the period to be pulled out, similar to the summary stats on the main page:
screen shot 2017-11-03 at 11 04 08 am

So it will be Average Duration: 4:00 Number of Sessions: 500.

These numbers will be for the entire period.

**Comment by jkelvie** [comment]
The chart looks good to me. I think this is ready to go functionally - @chris-ramon any comments on the code?

**Comment by jkelvie** [comment]
Initial notes:
1) Time period resets to zero when it goes over an hour - instead it should show "1h:10m:15s" for a period of "70:15" - you can see this below on the y-axis for average session duration

2) The x-axis goes from most recent to oldest - it should be the other way around.

screen shot 2017-11-21 at 3 06 04 pm

File: src/pages/audioplayerpage/AudioPlayerStats.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        }
+    }
+
+    async startLoading(props: AudioPlayerStatsProps): Promise<LogService.AudioPlayerStats> {
+        const query: Query = new Query();
+        query.add(new SourceParameter(props.source));
+        query.add(new StartTimeParameter(props.startDate));
+        query.add(new EndTimeParameter(props.endDate));
+        const [sessions, duration] = await Promise.all([LogService.getAudioSessions(query), LogService.getAudioDuration(query)]);
+        const audioSessions = sessions.map((audioSession: any) => {
```
Comment (chris-ramon) [inline src/pages/audioplayerpage/AudioPlayerStats.tsx]: @AnkRaiza could we move this operations: `sessions.map` and `audioSessions.reduce` to the API side.

File: src/pages/audioplayerpage/NumberSessions.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        if (!source) return;
+        this.mapState({ data: [], sourceState: 1});
+        const query: Query = new Query();
+        query.add(new SourceParameter(source));
+        query.add(new StartTimeParameter(this.props.startDate));
+        query.add(new EndTimeParameter(this.props.endDate));
+
+        let formatedData;
+        try {
+            const data = await AudioPlayerService.getAudioSessions(query);
```
Comment (chris-ramon) [inline src/pages/audioplayerpage/NumberSessions.tsx]: Also this part could we move it to the API side ?

File: src/pages/audioplayerpage/NumberSessions.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            return [];
+        }
+
+        const query: Query = new Query();
+        query.add(new SourceParameter(source));
+        query.add(new StartTimeParameter(startDate));
+        query.add(new EndTimeParameter(endDate));
+        try {
+            const data = await AudioPlayerService.getAudioSessions(query);
+            const audioSessions = data.map((audioSession: any) => {
```
Comment (chris-ramon) [inline src/pages/audioplayerpage/NumberSessions.tsx]: same here, could we move it to the API side ?

File: src/pages/audioplayerpage/SessionDuration.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        if (!source) return;
+        this.mapState({ data: [], sourceState: 1});
+        const query: Query = new Query();
+        query.add(new SourceParameter(source));
+        query.add(new StartTimeParameter(this.props.startDate));
+        query.add(new EndTimeParameter(this.props.endDate));
+
+        let formatedData;
+        try {
+            const duration = await AudioPlayerService.getAudioDuration(query);
```
Comment (chris-ramon) [inline src/pages/audioplayerpage/SessionDuration.tsx]: Also here, could we move it to the API side, as well as the operations within `startLoading`.

**Pull request merged**